vm/src/stdlib/os.rs

```diff
@@ -512,33 +577,128 @@ pub fn make_module(vm: &VirtualMachine) -> PyObjectRef {
          "st_size" => ctx.new_property(StatResultRef::st_size),
     });
 
-    py_module!(vm, "_os", {
-        "open" => ctx.new_rustfunc(os_open),
+    let fd_funcs: Vec<(&str, PyObjectRef, Option<bool>, Option<bool>, Option<bool>)> = vec![
```

Comment:
Why is the `support_*` an `Option`? This can be either `true` of `false`. You convert the `None` to `false` later

Reply:
It is a kind of documentation describing it is varies by platform(Some) or always not supported(None)

Reply:
I am not sure I understand...
Let's take `os.symlink` for example. on linux system `dir_fd` is supported and on windows it is not. In that case we will have `Some(true)` on linux and `Some(false)` on windows. While `follow_symlinks` will always be `None`. Is there any difference between `Some(false)` and `None`? Would there be a different error message somewhere I am missing?

Reply:
There is no difference between `None` and `Some(false)` for implementations. As you pointed out, `None` marks `os.symlink` doesn't have any option about `follow_symlinks`. I think it is a little bit confusing because it is just `Some(false)` for now.
Once we get actual values for each platforms, the `None` part always will be `None`. Because Python spec doesn't have any value about it. Otherwise, the `Some` part will be something like `Some(platform_supports_os_symlink)` because Python spec describes the value must be `true` or `false` by platforms.

Reply:
Is `platform_supports_os_symlink` a function in your example or a boolean value?

Reply:
It is a value. (or result of a function) I think the values are platform-specific constants defined in compile time.

Reply:
My opinion is that this `Option` is not needed. I think using `platform_supports_os_symlink` or `false` would be more readable.